src/Agent.Listener/Configuration/CredentialProvider.cs

```diff
@@ -64,7 +64,54 @@ public override void EnsureCredential(IHostContext context, CommandSettings comm
             trace.Info(nameof(EnsureCredential));
             ArgUtil.NotNull(command, nameof(command));
             CredentialData.Data[Constants.Agent.CommandLine.Args.Token] = command.GetToken();
-        }        
+        }
+    }
+
+    public sealed class ServiceIdentityCredential : CredentialProvider
+    {
+        public ServiceIdentityCredential(): base(Constants.Configuration.ServiceIdentity) {}
```

Comment:
What is the timeframe for supporting the new agent in MMS? Any reason we couldn't always use the new OAuth tokens instead of continuing to use ACS service identities?